### PR TITLE
Sequence package added as part of peerDepencies

### PR DESCRIPTION
**Pull request by SamueleA**
### Description
Made some dependencies peerDependencies:
`@web3-onboard/common`: Will be provided by `web3-onboard/core`, which is always installed prior to usage of the connector.

`0xsequence`: User will be able to select the version he wants to use and stay up-to-date with most recent release without having to always create a PR to update the sequence version.

### Checklist
- [x] The version field in `package.json` is incremented following [semantic versioning](https://semver.org/)
- [x] The box that allows repo maintainers to update this PR is checked
- [x] I tested locally to make sure this feature/fix works
- [x] I have run `yarn check` & `yarn build` to confirm there are not any associated errors
- [ ] This PR passes the Circle CI checks

### CODE REVIEW

**Comment by aaronbarnardsound** [comment]
Thanks @SamueleA, we'll take a look at this soon.

File: packages/sequence/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@web3-onboard/sequence",
-  "version": "2.0.2",
+  "version": "2.0.3",
```
Comment (Adamj1232) [inline packages/sequence/package.json:3]:
```suggestion
  "version": "2.0.3-alpha.1",
```

File: examples/with-vite-react/package.json [hunk 2]
```diff
@@ -21,7 +21,8 @@
     "@web3-onboard/mew": "^2.1.3",
     "@web3-onboard/portis": "^2.1.3",
     "@web3-onboard/react": "^2.3.5",
-    "@web3-onboard/sequence": "^2.0.2",
+    "@web3-onboard/sequence": "^2.0.3",
+    "@0xsequence": "^0.42.8",
```
Comment (Adamj1232) [inline examples/with-vite-react/package.json:25]: Are these changes required in the example projects?

**Review by Adamj1232 — CHANGES_REQUESTED** [review]
A few comments but should be able to get this in shortly, thanks!

File: packages/sequence/package.json (hunk 1 above)
Comment (SamueleA) [reply packages/sequence/package.json]: I made the change in commit 148699b

File: examples/with-vite-react/package.json (hunk 2 above)
Comment (SamueleA) [reply examples/with-vite-react/package.json:25]: The `@web3-onboard/sequence` is updated to match the updated version of the connector from this PR.

Because of the version update above, the `@0xsequence` dependency becomes necessary  since `@web3-onboard/sequence` includes it as a `peerDependency`, meaning it needs to be installed together with the connector.

Comment (Adamj1232) [reply examples/with-vite-react/package.json:25]: @SamueleA right but once the new sequence version is released this import will not be required as the sequence package will pull in the dependency correct?

File: examples/with-vite-react/package.json [hunk 3]
```diff
@@ -21,7 +21,8 @@
     "@web3-onboard/mew": "^2.1.3",
     "@web3-onboard/portis": "^2.1.3",
     "@web3-onboard/react": "^2.3.5",
-    "@web3-onboard/sequence": "^2.0.2",
+    "@web3-onboard/sequence": "^2.0.3",
```
Comment (Adamj1232) [inline examples/with-vite-react/package.json:24]:
```suggestion
    "@web3-onboard/sequence": "^2.0.3"-alpha.1,
```

File: examples/with-vite-react/package.json (hunk 2 above)
Comment (SamueleA) [reply examples/with-vite-react/package.json:25]: That's correct! My bad!

I removed the 0xSequence dependency

Comment (Adamj1232) [reply examples/with-vite-react/package.json:25]: @SamueleA great! Once you have that pushed I can test it out, thanks!

File: packages/sequence/package.json [hunk 4]
```diff
@@ -53,10 +53,12 @@
...
   "devDependencies": {
-    "typescript": "^4.5.5"
-  },
-  "dependencies": {
+    "typescript": "^4.5.5",
     "@web3-onboard/common": "^2.2.3",
     "0xsequence": "^0.40.5"
+  },
+  "peerDependencies": {
+    "@web3-onboard/common": ">=2.2.3",
```
Comment (Adamj1232) [inline packages/sequence/package.json:61]:
```suggestion
```
I think this can be removed as well

File: examples/with-vite-react/package.json (hunk 2 above)
Comment (Adamj1232) [reply examples/with-vite-react/package.json:25]:
```suggestion
```

File: packages/sequence/package.json [hunk 5]
```diff
@@ -53,10 +53,11 @@
   },
   "license": "MIT",
   "devDependencies": {
-    "typescript": "^4.5.5"
-  },
-  "dependencies": {
+    "typescript": "^4.5.5",
```
Comment (Adamj1232) [inline packages/sequence/package.json:56]:
```suggestion
    "typescript": "^4.5.5"
  },
  "dependencies": {
```

**Review by Adamj1232 — APPROVED** [review]
Fixed a few issues, its ready to go!
